sdk/identity/Azure.Identity/src/KubernetesProxyConfig.cs

```diff
@@ -0,0 +1,149 @@
+            {
+                throw new InvalidOperationException(
+                    "AZURE_KUBERNETES_TOKEN_PROXY environment variable is not set but other proxy-related " +
+                    "environment variables are present. This is likely a configuration issue.");
+            }
+
+            // Validate CA configuration is not ambiguous
+            if (!string.IsNullOrEmpty(caFile) && !string.IsNullOrEmpty(caData))
+            {
+                throw new InvalidOperationException(
+                    "Azure Kubernetes token proxy configuration is ambiguous. " +
+                    "Both AZURE_KUBERNETES_CA_FILE and AZURE_KUBERNETES_CA_DATA are set. " +
+                    "Please set only one of these environment variables.");
+            }
+
+            // Validate proxy URL format
+            if (!Uri.TryCreate(proxyUrl, UriKind.Absolute, out Uri parsedUrl))
+            {
+                throw new InvalidOperationException(
+                    $"AZURE_KUBERNETES_TOKEN_PROXY value '{proxyUrl}' is not a valid URL.");
+            }
+
+            // Validate HTTPS scheme
+            if (!string.Equals(parsedUrl.Scheme, "https", StringComparison.OrdinalIgnoreCase))
+            {
+                throw new InvalidOperationException(
+                    $"AZURE_KUBERNETES_TOKEN_PROXY must use HTTPS scheme. Got: {parsedUrl.Scheme}");
+            }
+
+            // Validate no user info, query, or fragment
+            if (!string.IsNullOrEmpty(parsedUrl.UserInfo))
+            {
+                throw new InvalidOperationException(
+                    "AZURE_KUBERNETES_TOKEN_PROXY must not contain user info.");
+            }
+
+            if (!string.IsNullOrEmpty(parsedUrl.Query))
+            {
+                throw new InvalidOperationException(
+                    "AZURE_KUBERNETES_TOKEN_PROXY must not contain query parameters.");
+            }
+
+            if (!string.IsNullOrEmpty(parsedUrl.Fragment))
+            {
+                throw new InvalidOperationException(
+                    "AZURE_KUBERNETES_TOKEN_PROXY must not contain a fragment.");
+            }
+
+            // Validate CA file exists if specified (content validation happens at use time)
+            if (!string.IsNullOrEmpty(caFile))
+            {
+                if (!File.Exists(caFile))
+                {
+                    throw new InvalidOperationException(
+                        $"AZURE_KUBERNETES_CA_FILE path '{caFile}' does not exist.");
+                }
+            }
+
+            // Basic validation for CA data if specified (detailed validation happens at use time)
+            if (!string.IsNullOrEmpty(caData))
+            {
+                if (string.IsNullOrWhiteSpace(caData))
+                {
+                    throw new InvalidOperationException(
+                        "AZURE_KUBERNETES_CA_DATA is empty or contains only whitespace.");
+                }
```

Comment:
This won't throw if caData is empty.
```suggestion
            if (string.IsNullOrEmpty(caData) || string.IsNullOrWhiteSpace(caData))
            {
                throw new InvalidOperationException(
                        "AZURE_KUBERNETES_CA_DATA is empty or contains only whitespace.");
            }
```